Alter dproj+dpks to make libs version agnostic (I hope)

peardox opened the pull request:
Analysis of the other dproj + dpk files in this project (i.e. P4D etc) I worked out what looked correct and tested it by wiping everything called P4DPyPackage* in /mnt/c/Users/Public/Documents/Embarcadero/Studio/21.0

Rebuilding this library recreated the properly named packages such as P4DPyPackage270.bpl (correct for CE ATM) and should also create P4DPyPackage280.bpl files for D11

Actually, this _should_ work for all versions of Delphi regardless of version (not that I can check anything other than CE of course)

lmbelo commented:
All packages should have the version property set to AUTO. let me check why it turned to a specific version in some moment.

peardox commented:
Hmm - looking at the diff Delphi's gone and converted all your 19.4s into 19.2s again - this started as a synched copy so had 19.4s...

Same for some of the othert stuff

I only actually made a few changes to replace the 280 stuff with Auto versions (and deleted 2x  hard-linked D11 delpoys)

lmbelo commented:
> > All packages should have the version property set to AUTO. let me check why it turned to a specific version in some moment.
> 
> They didn't - it was all 280 for this specific lib

Let's update it then.

peardox commented:
There's some Delphi IDE way of doing what I did manually I presume - fix it that way if possible

lmbelo commented on packages/AndroidManifest.template.xml:
```diff
@@ -0,0 +1,45 @@
+<?xml version="1.0" encoding="utf-8"?>
```
We can remove the Android manifest, it won't make effect in final apps.

peardox replied:
Fine by me, I was just testing it properly

lmbelo commented on packages/P4DPyPackage.dproj:
```diff
@@ -2,13 +2,13 @@
     <PropertyGroup>
```
We need to keep it all to Win32/Release by default.

peardox replied:
Oops - missed that bit - I don't use 32 bit anything (hardly anyone does in reality)

lmbelo replied:
We use Win32/Release because design-time packages are Win32 only.

peardox replied:
Oh, yeah - of course...

Are you supposed to install the design-times for all OSes? Sometimes things don't work if I don't (which is confusing)

Anyway - I'll be able to wipe Delphi again and try a clean system install for my tutorial soon :)

lmbelo replied:
Design-time packages are Win32 only, then we make it as default.

lmbelo replied:
if I skip other platforms from design-time packages, I can use the "Set active platform" option for project groups and the design-time packages will hold Win32 settings.

![Screen Shot 2022-08-09 at 11 29 18 1](https://user-images.githubusercontent.com/8376898/183675882-8704cd24-5d0a-4522-89c7-d7c3129bdf52.png)

peardox replied:
In the case of P4D Data Sciences in particular it would be far easier to manage if there was a branch that only contained design-time and another that contained run-time cos this is a mess...

![image](https://user-images.githubusercontent.com/1549219/183676296-cd750b1c-62e4-4145-bd8d-01503e8bc1a2.png)

I manually altered the dproj file to get my newer ones down the bottom and the DTPs/RTPs all bunched together (which is better than the mess you get by default)

The above could be improved of course

lmbelo replied:
Do you mean other project group?

peardox replied:
No, more tree nodes so everything wasn't mixed up as is the case usually

Can't work out how to do that though so just ordered them sensibly - at least I can group select with ease that way...

peardox replied:
The main Python4Delphi is horrible in this respect. Everything requires Python but it's not the first to be built. After thasn everything wants PythonEngine and that's not in order either.

I'm not certain what order you should build in but invariably just selecting build all ressults in something going wrong - not exactly easy to install...

lmbelo replied:
Weird, it should be working fine. Which project group are you using?

peardox replied:
Compile both for safety (never works first time)

I'll do a full wipe (also harder done than said) then install from scratch and record it all for Video and/or Grabs - it's the only good QA method.

Not doing that right now though - I'm all Delphi'd out ATM - need to switch to anything else for some hours to reset my brain

lmbelo commented:
We don't need to track resource files. We can remove all new .res from commits.

peardox commented:
Was just using your .gitignore

lmbelo commented:
> 



> There's some Delphi IDE way of doing what I did manually I presume - fix it that way if possible
![Screen Shot 2022-08-09 at 11 17 08](https://user-images.githubusercontent.com/8376898/183672627-180bc3bd-2e26-4ec1-9eee-5ca4b629a86f.png)

lmbelo commented:
> 

Let's update .gitignore to exclude .res and .template.xml

peardox commented:
> > 
> 
> > There's some Delphi IDE way of doing what I did manually I presume - fix it that way if possible
> > ![Screen Shot 2022-08-09 at 11 17 08](https://user-images.githubusercontent.com/8376898/183672627-180bc3bd-2e26-4ec1-9eee-5ca4b629a86f.png)

Ahh - I live and learn (pretty obvious when I think about it)

lmbelo commented:
> > > There's some Delphi IDE way of doing what I did manually I presume - fix it that way if possible
> > > ![Screen Shot 2022-08-09 at 11 17 08](https://user-images.githubusercontent.com/8376898/183672627-180bc3bd-2e26-4ec1-9eee-5ca4b629a86f.png)
> 
> Ahh - I live and learn (pretty obvious when I think about it)

It is only available in Delphi 10.4 and up.

peardox commented:
> > 
> 
> Let's update .gitignore to exclude .res and .template.xml. We should also replicate it to the others repos.

And lib/

lmbelo commented:
> > Let's update .gitignore to exclude .res and .template.xml. We should also replicate it to the others repos.
> 
> And lib/

Exactly.

lmbelo commented:
Update your clone and check if the latest changes works as expected.

peardox commented:
Line 291 is still wrong 
https://github.com/Embarcadero/Lightweight-Python-Wrappers/blob/main/packages/P4DPyPackage.dproj#L291

Line 148 is still wrong
https://github.com/Embarcadero/Lightweight-Python-Wrappers/blob/main/packages/dclP4DPyPackage.dproj#L148

lmbelo commented:
> Line 291 is still wrong https://github.com/Embarcadero/Lightweight-Python-Wrappers/blob/main/packages/P4DPyPackage.dproj#L291
> 
> Line 148 is still wrong https://github.com/Embarcadero/Lightweight-Python-Wrappers/blob/main/packages/dclP4DPyPackage.dproj#L148

It should update automatically once you open it in other version. Did you try to compile it?

lmbelo commented:
> > Line 291 is still wrong https://github.com/Embarcadero/Lightweight-Python-Wrappers/blob/main/packages/P4DPyPackage.dproj#L291
> > Line 148 is still wrong https://github.com/Embarcadero/Lightweight-Python-Wrappers/blob/main/packages/dclP4DPyPackage.dproj#L148
> 
> It should update automatically once you open it in other version. Did you try to compile it?

I found the issue. I'm fixing it.

peardox commented:
I'm in the middle of tests

Fortunately not got to this one yet

Here is a very basic set (marginally above hello world but checks it can load PSUtil + Torch) of VCL/FMX tests to make sure the full set work

https://github.com/peardox/P4D-Tests.git

lmbelo commented:
> I'm in the middle of tests - lemme know when you get to this one, I'll stop tests for now
> 
> Fortunately not got to this one yet
> 
> Here is a very basic set (marginally above hello world but checks it can load PSUtil + Torch) of VCL/FMX tests to make sure the full set work
> 
> https://github.com/peardox/P4D-Tests.git

Ah, that wasn't an issue at all, but I have upgraded project files.
Let's try again.

peardox commented:
I'm testing using the following method.
Remove all traces of python stuff in (fortunately that's all these contain anyway)....
C:\Users\Public\Documents\Embarcadero\Studio\21.0\Dcp
C:\Users\Public\Documents\Embarcadero\Studio\21.0\Bpl
(21.0 will be 22.0 for you)

Install packages in correct order - 4 steps of...

p4d = Install VCL package  + components for windows (fmx will be used for later tests for Android + Mac), for others = Install package + components 
First pass - install W32 release + debug then w64 release + debug, later passes Android / Mac
Add the debug lib path for OS to options
Assuming no compile issues...
See if https://github.com/peardox/P4D-Tests.git run (inv VCL for Win, FMX only for others)

lmbelo commented:
> I'm testing using the following method. Remove all traces of python stuff in (fortunately that's all these contain anyway).... C:\Users\Public\Documents\Embarcadero\Studio\21.0\Dcp C:\Users\Public\Documents\Embarcadero\Studio\21.0\Bpl (21.0 will be 22.0 for you)
> 
> Install packages in correct order - 4 steps of...
> 
> p4d = Install VCL package + components for windows (fmx will be used for later tests for Android + Mac), for others = Install package + components First pass - install W32 release + debug then w64 release + debug, later passes Android / Mac Add the debug lib path for OS to options Assuming no compile issues... See if https://github.com/peardox/P4D-Tests.git run (inv VCL for Win, FMX only for others)

It worth mentioning that you can use the [P4DComponentSuite](https://github.com/Embarcadero/python4delphi/tree/master/Packages/Delphi) for a complete compilation.

lmbelo commented:
> > I'm testing using the following method. Remove all traces of python stuff in (fortunately that's all these contain anyway).... C:\Users\Public\Documents\Embarcadero\Studio\21.0\Dcp C:\Users\Public\Documents\Embarcadero\Studio\21.0\Bpl (21.0 will be 22.0 for you)
> > Install packages in correct order - 4 steps of...
> > p4d = Install VCL package + components for windows (fmx will be used for later tests for Android + Mac), for others = Install package + components First pass - install W32 release + debug then w64 release + debug, later passes Android / Mac Add the debug lib path for OS to options Assuming no compile issues... See if https://github.com/peardox/P4D-Tests.git run (inv VCL for Win, FMX only for others)
> 
> It worth mentioning that you can use the [P4DComponentSuite](https://github.com/Embarcadero/python4delphi/tree/master/Packages/Delphi) for a complete compilation.

I need to switch back to the Python debugger for Android. We are producing some content along the DelphiFMXBuilder for the coming boot camp. Maybe I'd delay to reply, but I'll do my best.
There are some small enhancements that I'd like to introduce to the whole Python projects ecosystem. I will work out on it as possible.

peardox commented:
> > I'm testing using the following method. Remove all traces of python stuff in (fortunately that's all these contain anyway).... C:\Users\Public\Documents\Embarcadero\Studio\21.0\Dcp C:\Users\Public\Documents\Embarcadero\Studio\21.0\Bpl (21.0 will be 22.0 for you)
> > Install packages in correct order - 4 steps of...
> > p4d = Install VCL package + components for windows (fmx will be used for later tests for Android + Mac), for others = Install package + components First pass - install W32 release + debug then w64 release + debug, later passes Android / Mac Add the debug lib path for OS to options Assuming no compile issues... See if https://github.com/peardox/P4D-Tests.git run (inv VCL for Win, FMX only for others)
> 
> It worth mentioning that you can use the [P4DComponentSuite](https://github.com/Embarcadero/python4delphi/tree/master/Packages/Delphi) for a complete compilation.

Caused issues in the past when installing non-Windows - I'll get to that in a bit, just finishing off Windows installation.

P4D Data Science - fiddly component install...

![image](https://user-images.githubusercontent.com/1549219/183753621-5364195c-4d38-4430-868b-0b4e0db3a78d.png)

This is using current repo - notice you have to click on every OTHER lib and make sure they all start dcl... This is error prone and the reason I put the DTPs in a bunch at the end - as this way click on first, shift click on last, Right Click install (much more developer friendly)

There are currently 19 components and that number will grow. They either need splitting up of bunching as I've done. Problem with splitting up is that unless handled carefully you'll end up with a really long list of paths to add to options

ATM I'm just adding e.g. C:\DelphiComponents\P4D-Data-Sciences\lib\Win32\Debug

I'd prefer something like C:\DelphiComponents\P4D-Data-Sciences\${PLATFORM)\$(Config) but that didn't work

peardox commented:
> I need to switch back to the Python debugger for Android. We are producing some content along the DelphiFMXBuilder for the coming boot camp. Maybe I'd delay to reply, but I'll do my best. There are some small enhancements that I'd like to introduce to the whole Python projects ecosystem. I will work out as possible.

Saw boot camp thing - a few topics might be of interest so just signed up

lmbelo commented:
> > > I'm testing using the following method. Remove all traces of python stuff in (fortunately that's all these contain anyway).... C:\Users\Public\Documents\Embarcadero\Studio\21.0\Dcp C:\Users\Public\Documents\Embarcadero\Studio\21.0\Bpl (21.0 will be 22.0 for you)
> > > Install packages in correct order - 4 steps of...
> > > p4d = Install VCL package + components for windows (fmx will be used for later tests for Android + Mac), for others = Install package + components First pass - install W32 release + debug then w64 release + debug, later passes Android / Mac Add the debug lib path for OS to options Assuming no compile issues... See if https://github.com/peardox/P4D-Tests.git run (inv VCL for Win, FMX only for others)
> > 
> > 
> > It worth mentioning that you can use the [P4DComponentSuite](https://github.com/Embarcadero/python4delphi/tree/master/Packages/Delphi) for a complete compilation.
> 
> Caused issues in the past when installing non-Windows - I'll get to that in a bit, just finishing off Windows installation.
> 
> P4D Data Science - fiddly component install...
> 
> ![image](https://user-images.githubusercontent.com/1549219/183753621-5364195c-4d38-4430-868b-0b4e0db3a78d.png)
> 
> This is using current repo - notice you have to click on every OTHER lib and make sure they all start dcl... This is error prone and the reason I put the DTPs in a bunch at the end - as this way click on first, shift click on last, Right Click install (much more developer friendly)
> 
> There are currently 19 components and that number will grow. They either need splitting up of bunching as I've done. Problem with splitting up is that unless handled carefully you'll end up with a really long list of paths to add to options
> 
> ATM I'm just adding e.g. C:\DelphiComponents\P4D-Data-Sciences\lib\Win32\Debug
> 
> I'd prefer something like C:\DelphiComponents\P4D-Data-Sciences\${PLATFORM)\$(Config) but that didn't work

I will set BOSS as the IDE package installer, then we would not need to follow up all these installation steps.

BTW, using environment variables to set up paths works fine.

peardox commented:
The VCL test project looks like it's running (Torch install - long wait...)

peardox commented:
Tried env vars, $(Release) seemed OK but $(Config) wan't liked (greyed out in IDE)
I guess lib\\$(Release)\Debug is ok though

peardox commented:
> I will set BOSS as the IDE package installer - I'll look it up

Anyway - tests ran as correctly for both VCL + FLM under Win64 so that stage of testing is complete.

Mac and Android are another issue (extremely problematic in the past)

lmbelo commented:
> > I will set BOSS as the IDE package installer - I'll look it up
> 
> Anyway - tests ran as correctly for both VCL + FLM under Win64 so that stage of testing is complete.
> 
> Mac and Android are another issue (extremely problematic in the past)

It should be very straight forward to get it working on Android and Mac. Let me know if you have any issue.
